jina/peapods/remote.py

```diff
@@ -58,9 +65,9 @@ def delete_remote(self):
         if hasattr(self, 'api') and self.api.is_alive and self.remote_id:
             self.api.delete(self.remote_id)
 
-    def close(self):
-        self.delete_remote()
-        self.join()
+    def close(self) -> None:
+        self.send_terminate_signal()
+        # TODO: I would rather wait for loop_body to finish but it seems to go on forever
```

Comment:
Do we intend on waiting for timeout from Send terminate request for exit instead of waiting forever? We could adjust the timeout experimentally around the time taken by loop body to ideally exit.
We are injecting the timeout info in the first pea

Reply:
the timeout is to sent the signal (it is a connection parameter), but right now we do not have any info of reply back knowing that something has been properly closed